tests/nativeapp/test_manager.py

```diff
@@ -1353,32 +1353,85 @@ def test_account_event_table_not_set_up(mock_execute, temp_dir, mock_cursor):
     assert native_app_manager.account_event_table == ""
 
 
+@pytest.mark.parametrize(
+    [
+        "since",
+        "expected_since_clause",
+        "until",
+        "expected_until_clause",
+        "limit",
+        "expected_limit_clause",
+    ],
+    [
+        (
+            "",
+            "",
+            "",
+            "",
+            0,
+            "",
+        ),
+        (
+            "1 hour",
+            "and timestamp >= sysdate() - interval '1 hour'",
+            "20 minutes",
+            "and timestamp <= sysdate() - interval '20 minutes'",
+            0,
+            "",
+        ),
+        ("", "", "", "", 10, "limit 10"),
+        (
+            "1 hour",
+            "and timestamp >= sysdate() - interval '1 hour'",
+            "20 minutes",
+            "and timestamp <= sysdate() - interval '20 minutes'",
+            10,
+            "limit 10",
+        ),
+    ],
+)
 @mock.patch(
     NATIVEAPP_MANAGER_ACCOUNT_EVENT_TABLE,
     return_value="db.schema.event_table",
     new_callable=mock.PropertyMock,
 )
 @mock.patch(NATIVEAPP_MANAGER_EXECUTE)
-def test_get_events(mock_execute, mock_account_event_table, temp_dir, mock_cursor):
+def test_get_events(
+    mock_execute,
+    mock_account_event_table,
+    temp_dir,
+    mock_cursor,
+    since,
+    expected_since_clause,
+    until,
+    expected_until_clause,
+    limit,
+    expected_limit_clause,
+):
     create_named_file(
         file_name="snowflake.yml",
         dir_name=temp_dir,
         contents=[mock_snowflake_yml_file],
     )
 
+    events = [dict(TIMESTAMP="2020-01-01T00:00:00Z", VALUE="test")] * 100
```

Comment:
the actual data doesn't matter since we don't run the SQL for real, but I still wanted a meaningful amount of rows in the response